pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -117,16 +117,6 @@ type GraphBuilder struct {
 type monitor struct {
 	controller cache.Controller
 	store      cache.Store
-
-	// stopCh stops Controller. If stopCh is nil, the monitor is considered to be
-	// not yet started.
-	stopCh chan struct{}
-}
-
-// Run is intended to be called in a goroutine. Multiple calls of this is an
-// error.
-func (m *monitor) Run() {
-	m.controller.Run(m.stopCh)
```

Comment:
m.controller.Run(...) still needs to be called, no?

Reply:
No, this actually never did anything in the first place.

Previously, it was believed that calling sharedInformer.GetController() would return a real controller that could be ran. https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/garbagecollector/graph_builder.go#L177

but in reality, it just returns a dummy controller that does a no-op https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/shared_informer.go#L452

Hence we needed to modify the informer to shut itself down (via the stopoptions) since there is no way to access the controller of an informer.